enhancements/microshift/etcd-supportability.md

```diff
@@ -0,0 +1,249 @@
+  - None
+creation-date: 2023-03-02
+last-updated: 2023-03-02
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com//browse/ETCD-356
+see-also:
+  - "/enhancements/microshift/etcd-as-a-transient-systemd-unit.md"
+  - "/enhancements/microshift/periodic-etcd-defragmentation-in-microshift.md"
+  - "/enhancements/microshift/update-and-rollback.md"
+replaces:
+  - "https://github.com/openshift/enhancements/pull/1356"
+---
+
+# Supporting etcd as used in MicroShift
+
+## Summary
+
+This enhancement captures some decisions about how we will provide
+support procedures and tools for etcd as it is configured and used in
+MicroShift deployements.
+
+## Motivation
+
+We need to decide what approach we will take for support procedures
+and tools for the etcd database used by MicroShift. We need to
+consider how closely those procedures should be like the procedures
+for OCP, and whether it makes sense to try to reuse the same tools. We
+have generally taken the stance that we want to align, but not at the
+expense of the user experience or the ability to meet the use cases
+for which MicroShift is being developed.
+
+### User Stories
+
+* As a MicroShift administrator, I want to back up the database used
+  by MicroShift so that I can restore it if something goes wrong with
+  the host.
+* As a MicroShift administrator, I want to restore the MicroShift
+  database when something goes wrong to reset the host to a known good
+  state.
+* As a MicroShift administrator, I want to fix the certificates used
+  to communicate with the database in case something goes wrong.
+
+### Goals
+
+* Describe the relationship between MicroShift and etcd and how it
+  differs from OpenShift.
+* Outline the support procedures for the user stories listed above.
+
+### Non-Goals
+
+* This document will not contain complete support procedures. More
+  complete documentation will be written separately.
+* MicroShift does not own the workloads running on it. Backing up and
+  restoring the state of applications running on MicroShift is out of
+  scope.
+
+## Proposal
+
+### Design Considerations
+
+MicroShift is designed to provide a unified user experience. Many
+components, including the kubernetes API server, kubelet, and
+controllers, are compiled into a single binary. Users interact with
+MicroShift, rather than the individual components.
+
+MicroShift is not a “full stack” solution. It does not manage the OS
+on which it runs, and is meant to integrate with, rather than subsume,
+tools and processes for managing the host. “MicroShift is an
+application running on RHEL.”
+
+MicroShift is meant to run on edge devices, where manual operations
+are not always possible. It should therefore be as resilient as
+possible and take automated action when possible.
```

Comment:
That's another discussion, however, do we want fully automated (operators like) semi-automated (ansible playbooks and scripts in images)?

Reply:
Good question. We'll have different levels of automation for different things. In this document, for example, we mention automating backup and restore during greenboot events and automating the defragmentation of the database. However, automating user-managed backups is not part of the work we have planned.